include/IECoreDelight/ShaderNetworkAlgo.h

```diff
@@ -48,6 +49,30 @@ namespace ShaderNetworkAlgo
 
 IECOREDELIGHT_API IECoreScene::ShaderNetworkPtr preprocessedNetwork( const IECoreScene::ShaderNetwork *shaderNetwork );
 
+enum class USDLightType
+{
+	None,
+	SphereLight,
+	RectLight,
+	DiskLight,
+	DistantLight,
+	DomeLight,
+	CylinderLight
+};
+
+USDLightType usdLightType( const IECoreScene::ShaderNetwork *shaderNetwork );
+DelightHandleSharedPtr createUSDLightGeometry(
+	const DelightHandle &transformHandle,
+	const IECoreScene::ShaderNetwork *shaderNetwork,
+	const USDLightType usdLightType
```

Comment:
Passing both `USDLightType` _and_ `shaderNetwork` is redundant, since the former can be derived from the latter. It's a little bit interesting that it's not the USD type that actually matters either - it's the type of the NSI node that we're making. It's a pity it's not possible to query the type of a node from NSI, as I think that would make this whole thing a bit cleaner.

Reply:
Not really relevant any more after the refactor in https://github.com/GafferHQ/gaffer/pull/5603/commits/a0c61d62e4bb80bef1f2ddfe3839f68389fcb36d, now that we're not using `USDLightType`.